backend/de.metas.contracts/src/main/java/de/metas/contracts/interceptor/M_InOutLine.java

```diff
@@ -66,4 +45,14 @@ public void updateFLatrateDataEntryQtySubstract(final I_M_InOutLine doc)
 				doc, // inOutLine
 				true); // subtract = true
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE }, ifColumnsChanged = { org.compiere.model.I_M_InOutLine.COLUMNNAME_MovementQty })
+	public void verifyIfPartOfInterimContract(final org.compiere.model.I_M_InOutLine inOutLine)
```

Comment:
on DELETE don't we have to do something?

Reply:
also, is it OK do to things even if the input is draft?

Reply:
> also, is it OK do to things even if the input is draft?

no; I'm adding a comment at InterimInvoiceFlatrateTermBL.java

Reply:
> on DELETE don't we have to do something?

actually, imo we need to looks at the inout's docactions and then iterate its lines. like, if an inout is reversed..